airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/components/debezium/StateFilesAccessor.kt

```diff
@@ -0,0 +1,192 @@
+        val builder = ImmutableList.builder<HistoryRecord>()
+        val consumer = Consumer { element: HistoryRecord -> builder.add(element) }
+        fileSchemaHistory.start()
+        try {
+            RECOVER_RECORDS_METHOD.invoke(fileSchemaHistory, consumer)
+        } catch (e: IllegalAccessException) {
+            throw RuntimeException(e)
+        } catch (e: InvocationTargetException) {
+            throw RuntimeException(e)
+        }
+        fileSchemaHistory.stop()
+        return DebeziumComponent.State.Schema(builder.build())
+    }
+
+    fun writeSchema(schema: DebeziumComponent.State.Schema) {
+        fileSchemaHistory.initializeStorage()
+        fileSchemaHistory.start()
+        for (r in schema.debeziumSchemaHistory) {
+            try {
+                STORE_RECORD_METHOD.invoke(fileSchemaHistory, r)
+            } catch (e: IllegalAccessException) {
+                throw RuntimeException(e)
+            } catch (e: InvocationTargetException) {
+                throw RuntimeException(e)
+            }
+        }
+        fileSchemaHistory.stop()
+    }
+
+    init {
+        try {
+            workingDir = Files.createTempDirectory(Path.of("/tmp"), "airbyte-debezium-state")
+        } catch (e: IOException) {
+            throw UncheckedIOException(e)
+        }
+        offsetFilePath = workingDir.resolve("offset.dat")
+        schemaFilePath = workingDir.resolve("dbhistory.dat")
+        // Create and configure FileOffsetBackingStore instance.
+        fileOffsetBackingStore = FileOffsetBackingStore(keyConverter())
+        val offsetProps = HashMap<String, String>()
+        offsetProps[WorkerConfig.KEY_CONVERTER_CLASS_CONFIG] = JsonConverter::class.java.name
+        offsetProps[WorkerConfig.VALUE_CONVERTER_CLASS_CONFIG] = JsonConverter::class.java.name
+        offsetProps[StandaloneConfig.OFFSET_STORAGE_FILE_FILENAME_CONFIG] =
+            offsetFilePath.toString()
+        fileOffsetBackingStore.configure(StandaloneConfig(offsetProps))
+        // Create and configure FileSchemaHistory instance.
+        fileSchemaHistory = FileSchemaHistory()
+        val schemaConfig =
+            Configuration.create()
+                .with(FileSchemaHistory.FILE_PATH, schemaFilePath.toString())
+                .build()
+        fileSchemaHistory.configure(
+            schemaConfig,
+            HistoryRecordComparator.INSTANCE,
+            SchemaHistoryListener.NOOP,
+            false
+        )
+    }
+
+    companion object {
+        // Use reflection to access the necessary protected methods in FileSchemaHistory.
+        private val STORE_RECORD_METHOD: Method
+        private val RECOVER_RECORDS_METHOD: Method
+
+        init {
+            try {
+                STORE_RECORD_METHOD =
+                    AbstractFileBasedSchemaHistory::class
+                        .java
+                        .getDeclaredMethod("storeRecord", HistoryRecord::class.java)
+                RECOVER_RECORDS_METHOD =
```

Comment:
how is this done today?

Reply:
We don't do this at all, this parsing of the schema history file contents is new.  I thought it would be cute to expose the history at least for debugging purposes.